llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -2009,9 +2012,9 @@ static VPRecipeBase *createEVLRecipe(VPValue *HeaderMask,
       .Case<VPWidenSelectRecipe>([&](VPWidenSelectRecipe *Sel) {
         SmallVector<VPValue *> Ops(Sel->operands());
         Ops.push_back(&EVL);
-        return new VPWidenIntrinsicRecipe(Intrinsic::vp_select, Ops,
-                                          TypeInfo.inferScalarType(Sel),
-                                          Sel->getDebugLoc());
+        return new VPWidenIntrinsicRecipe(
+            Intrinsic::vp_select, Ops, TypeInfo.inferScalarType(Sel),
+            Sel->getMetadata(), Sel->getDebugLoc());
```

Comment:
Here and below VPWidenIntrinsicRecipe is relieved from receiving the underlying instruction `Sel`, so its metadata needs to be passed.